vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -74,37 +47,76 @@ ChakraRuntime::ChakraRuntime(ChakraRuntimeArgs &&args) noexcept : m_args{std::mo
+  SetCurrentContext(m_context);
 
   startDebuggingIfNeeded();
 
   setupNativePromiseContinuation();
 
   std::call_once(s_runtimeVersionInitFlag, initRuntimeVersion);
 
-  static facebook::jsi::StringBuffer bootstrapBundleSourceBuffer{g_bootstrapBundleSource};
-  evaluateJavaScriptSimple(bootstrapBundleSourceBuffer, "ChakraRuntime_bootstrap.bundle");
+  m_propertyId.Object = JsRefHolder{GetPropertyIdFromName(L"Object")};
+  m_propertyId.Proxy = JsRefHolder{GetPropertyIdFromName(L"Proxy")};
+  m_propertyId.Symbol = JsRefHolder{GetPropertyIdFromName(L"Symbol")};
+  m_propertyId.byteLength = JsRefHolder{GetPropertyIdFromName(L"byteLength")};
+  m_propertyId.configurable = JsRefHolder{GetPropertyIdFromName(L"configurable")};
+  m_propertyId.enumerable = JsRefHolder{GetPropertyIdFromName(L"enumerable")};
+  m_propertyId.get = JsRefHolder{GetPropertyIdFromName(L"get")};
+  m_propertyId.hostFunctionSymbol = JsRefHolder{GetPropertyIdFromSymbol(L"hostFunctionSymbol")};
+  m_propertyId.hostObjectSymbol = JsRefHolder{GetPropertyIdFromSymbol(L"hostObjectSymbol")};
+  m_propertyId.length = JsRefHolder{GetPropertyIdFromName(L"length")};
+  m_propertyId.message = JsRefHolder{GetPropertyIdFromName(L"message")};
+  m_propertyId.ownKeys = JsRefHolder{GetPropertyIdFromName(L"ownKeys")};
+  m_propertyId.propertyIsEnumerable = JsRefHolder{GetPropertyIdFromName(L"propertyIsEnumerable")};
+  m_propertyId.prototype = JsRefHolder{GetPropertyIdFromName(L"prototype")};
+  m_propertyId.set = JsRefHolder{GetPropertyIdFromName(L"set")};
+  m_propertyId.toString = JsRefHolder{GetPropertyIdFromName(L"toString")};
+  m_propertyId.value = JsRefHolder{GetPropertyIdFromName(L"value")};
+  m_propertyId.writable = JsRefHolder{GetPropertyIdFromName(L"writable")};
+
+  m_undefinedValue = JsRefHolder{GetUndefinedValue()};
 }
 
 ChakraRuntime::~ChakraRuntime() noexcept {
+  m_undefinedValue = {};
+  m_propertyId = {};
+  m_proxyConstructor = {};
+  m_hostObjectProxyHandler = {};
+
   stopDebuggingIfNeeded();
 
-  VerifyChakraErrorElseThrow(JsSetCurrentContext(JS_INVALID_REFERENCE));
-  m_context.Invalidate();
+  m_context = {};
+  SetCurrentContext(m_prevContext);
+  m_prevContext = {};
 
   JsSetRuntimeMemoryAllocationCallback(m_runtime, nullptr, nullptr);
 
-  VerifyChakraErrorElseThrow(JsDisposeRuntime(m_runtime));
+  ChakraVerifyJsErrorElseThrow(JsDisposeRuntime(m_runtime));
+}
+
+JsValueRef ChakraRuntime::CreatePropertyDescriptor(JsValueRef value, PropertyAttibutes attrs) {
+  JsValueRef descriptor = CreateObject();
+  SetProperty(descriptor, m_propertyId.value, value);
+  if (!(attrs & PropertyAttibutes::ReadOnly)) {
+    SetProperty(descriptor, m_propertyId.writable, BoolToBoolean(true));
+  }
+  if (!(attrs & PropertyAttibutes::DontEnum)) {
+    SetProperty(descriptor, m_propertyId.enumerable, BoolToBoolean(true));
+  }
+  if (!(attrs & PropertyAttibutes::DontDelete)) {
+    SetProperty(descriptor, m_propertyId.configurable, BoolToBoolean(true));
+  }
```

Comment:
writeable/readonly and dontenum/enumeable make sense as antonyms but... what does configurable/dontdelete mean?

Reply:
The 'configurable' is the term used by JavaScript. Which literally mean if the property can be deleted or not. Thus, I used the term borrowed from the JSC API: [https://docs.microsoft.com/en-us/dotnet/api/javascriptcore.jspropertyattributes?view=xamarin-ios-sdk-12](https://docs.microsoft.com/en-us/dotnet/api/javascriptcore.jspropertyattributes?view=xamarin-ios-sdk-12)

---
In reply to: [470466243](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470466243) [](ancestors = 470466243)